### PR TITLE
bugfix: fix the issue of missing sentinel password in store redis mode

### LINKED ISSUE #4161: Seata接入Redis哨兵集群失败
- [ ] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
Seata1.4.2版本，接入Redis哨兵集群的时候，假如哨兵节点配置密码的话，jedis客户端接入失败，报权限认证失败的错误。jedis客户端不支持sentinel节点设置密码,JedisSentinelPool 类中，74行 sentinelPassword 项默认为null导致的。
假如将客户端改为spring-boot-starter-data-redis， 2.3.4.RELEASE版本以上可配置sentinelPassword

### Ⅱ. Describe what happened

  If there is an exception, please attach the exception trace:

```
Just paste your stack trace here!
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

1. xxx
2. xxx
3. xxx

### Ⅵ. Environment:

- JDK version :
- Seata version:  
- OS :
- Others:

### PR DESCRIPTION
**Pull request by PeppaO**
- [ ] I have registered the PR [changes](../changes).

### Ⅰ. Describe what this PR did
当store.redis.mode=sentinel时，设置哨兵密码时，代码中缺少哨兵密码

### Ⅱ. Does this pull request fix one issue?

Fixes #4161

### CODE REVIEW

File: server/src/main/java/io/seata/server/storage/redis/JedisPooledFactory.java [hunk 1]
```diff
@@ -27,11 +27,7 @@
 import io.seata.core.constants.ConfigurationKeys;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import redis.clients.jedis.Jedis;
-import redis.clients.jedis.JedisPool;
-import redis.clients.jedis.JedisPoolAbstract;
-import redis.clients.jedis.JedisPoolConfig;
-import redis.clients.jedis.JedisSentinelPool;
+import redis.clients.jedis.*;
```
Comment (funky-eyes) [inline server/src/main/java/io/seata/server/storage/redis/JedisPooledFactory.java:30]: 不要使用*引入

File: server/src/main/java/io/seata/server/storage/redis/JedisPooledFactory.java [hunk 2]
```diff
@@ -98,7 +94,8 @@ public static JedisPoolAbstract getJedisPoolInstance(JedisPoolAbstract... jedisP
                             Set<String> sentinels = new HashSet<>(SENTINEL_HOST_NUMBER);
                             String[] sentinelHosts = CONFIGURATION.getConfig(ConfigurationKeys.STORE_REDIS_SENTINEL_HOST).split(",");
                             Arrays.asList(sentinelHosts).forEach(sentinelHost -> sentinels.add(sentinelHost));
-                            tempJedisPool = new JedisSentinelPool(masterName, sentinels, poolConfig, 60000, password, CONFIGURATION.getInt(ConfigurationKeys.STORE_REDIS_DATABASE, DATABASE));
+                            tempJedisPool = new JedisSentinelPool(masterName, sentinels, poolConfig, 60000, 60000, password, CONFIGURATION.getInt(ConfigurationKeys.STORE_REDIS_DATABASE, DATABASE),
+                                    null, Protocol.DEFAULT_TIMEOUT, Protocol.DEFAULT_TIMEOUT, CONFIGURATION.getConfig(ConfigurationKeys.STORE_REDIS_SENTINEL_PASSWORD), null);
```
Comment (funky-eyes) [inline server/src/main/java/io/seata/server/storage/redis/JedisPooledFactory.java:98]: 这后面的null是什么参数?

File: script/config-center/config.txt
Comment (funky-eyes) [inline script/config-center/config.txt:1]: 不止这里,server的yml里的配置要补充,以及starter模块里对应的properties也要补充

File: server/src/main/java/io/seata/server/storage/redis/JedisPooledFactory.java (hunk 2 above)
Comment (PeppaO) [reply server/src/main/java/io/seata/server/storage/redis/JedisPooledFactory.java:98]: clientName、sentinelClientName

Comment (funky-eyes) [reply server/src/main/java/io/seata/server/storage/redis/JedisPooledFactory.java:98]: ok

File: server/src/main/java/io/seata/server/storage/redis/JedisPooledFactory.java (hunk 1 above)
Comment (PeppaO) [reply server/src/main/java/io/seata/server/storage/redis/JedisPooledFactory.java]: done

File: script/config-center/config.txt
Comment (PeppaO) [reply script/config-center/config.txt:1]: done

**Review by funky-eyes — APPROVED** [review]
LGTM

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**